src/object-store/src/layers/lru_cache/read_cache.rs

```diff
@@ -211,11 +200,11 @@ pub struct ReadCacheReader<I: Access> {
 
 impl<I: Access> ReadCacheReader<I> {
     /// TODO: we can return the Buffer directly to avoid another read from cache.
-    async fn read_remote(&self, offset: u64, limit: usize) -> Result<ReadResult> {
+    async fn read_remote(&mut self) -> Result<ReadResult> {
         OBJECT_STORE_LRU_CACHE_MISS.inc();
 
-        let buf = self.inner_reader.read_at(offset, limit).await?;
-        let result = self.try_write_cache(buf, offset).await;
+        let buf = self.inner_reader.read_all().await?;
```

Comment:
So, instead of chunked read, we `read_all` the object here. It may cause some regressions like reading a large object can be slow, worse concurrency and worse cache friendly.

@Xuanwo told me that https://github.com/apache/opendal/issues/4672 is for this case. So I leave this to him for any further improvement.

@WenyXu You may test if we suffer some significant regression here.

Reply:
Should be addressed in https://github.com/tisonkun/greptimedb/pull/7

Reply:
Thank you!

Reply:
> @WenyXu You may test if we suffer some significant regression here.

The original design is cache the whole file. I think we may remove this cache layer ASAP. cc @evenyag

Reply:
> The original design is cache the whole file. I think we may remove this cache layer ASAP. cc @evenyag

Hi, I didn't get the enough context here. Our current implemention should be exact same with 0.6 version.

Reply:
> > @WenyXu You may test if we suffer some significant regression here.
> 
> The original design is cache the whole file. I think we may remove this cache layer ASAP. cc @evenyag

The original design is to cache the chunk.

Reply:
> Hi, I didn't get the enough context here. Our current implemention should be exact same with 0.6 version.

> The original design is to cache the chunk.

Sorry, I misremembered. The original design is chunk caching.